frontend/src/utils/websockets.js

```diff
@@ -43,25 +44,32 @@ export function connectWS(topic, onMessageCallback) {
 
   const connect = () => {
     let reconnectCounter = 0;
+    let badGatewayStatus = false;
     this.sockClient = new Client({
       brokerURL: config.WS_URL,
       debug(strMessage) {
         // console.log('debug: ', strMessage);
         // -- detect reconnect and increment the reconnect counter
         if (strMessage.includes('reconnect')) {
           getUserSession()
-            .then(({ data }) => {
+            .then((userSessionResp) => {
+              const { data, status } = userSessionResp;
               reconnectCounter =
                 data && !data.loggedIn ? reconnectCounter + 1 : 0;
+
+              if (status === 502) badGatewayStatus = true;
             })
             .catch(() => {
               reconnectCounter += 1;
             });
         }
+        // update the store flag
+        badGatewayStatus && store.dispatch(badGateway(BAD_GATEWAY_ERROR));
+
         // -- if more than max allowed reconnect times  ->  deactivate
         if (reconnectCounter > maxReconnectTimesNo) {
           this.reconnectDelay = 0; // 0 - deactivates the sockClient
-          store.dispatch(noConnection(true));
+          !badGatewayStatus && store.dispatch(noConnection(true));
```

Comment:
as wrote above... IMHO we shall unify different connection errors into one single event.
The actual reason why there is a connection error (e.g. bad gateway or internet down) can be a type.